### DISCUSSION TITLE
Closing Stream in provided demo

### DISCUSSION THREAD

**Discussion by robot-ranger** [mtconnect/cppagent#504] [Q&A]
what is the proper way to abort/close the stream in the [demo code](https://github.com/mtconnect/cppagent/blob/main/demo/twin/lib/mtconnect/protocol.js)? In our react app, I would like to close the stream when the components un-mount (ex user navigates away).

I tried calling `controller.abort()`. I get an `AbortError` exception, which i suppose is expected, but it seems like data continues to arrive.

I continue getting the console print on this line:https://github.com/mtconnect/cppagent/blob/d6bc71040f84868fb235c474f7bf716ae02477b8/demo/twin/lib/mtconnect/protocol.js#L119

Even after running `controller.abort` and navigating away. It doesnt appear to have actually aborted?

**Comment by robot-ranger** [comment]
I have verified that `controller.abort()` has run, i have set a break point and verified `aborted=true`. I also have a log print `console.log("component unmount and cleanup")`

questioning if i am not using the abort controller correctly

**Comment by robot-ranger** [comment]
@wsobel - any input here? I cant seem to figure out how to close a running stream programmatically.

**Comment by wsobel** [comment]
It looks like the protocol may keep running. We may need to put a flag in to stop the loop and exit streaming.
